src/CounterLabel/CounterLabel.tsx

```diff
@@ -1,47 +1,49 @@
-import React from 'react'
+import React, {HTMLAttributes, forwardRef} from 'react'
 import Box from '../Box'
 import {BetterSystemStyleObject, SxProp, merge} from '../sx'
 import VisuallyHidden from '../_VisuallyHidden'
+import {defaultSxProp} from '../utils/defaultSxProp'
 
-export type CounterLabelProps = {
-  scheme?: 'primary' | 'secondary'
-} & SxProp
+export type CounterLabelProps = React.PropsWithChildren<
+  HTMLAttributes<HTMLSpanElement> & {
+    scheme?: 'primary' | 'secondary'
+  } & SxProp
+>
 
-const CounterLabel: React.FC<React.PropsWithChildren<CounterLabelProps>> = ({
-  scheme = 'secondary',
-  sx = {},
-  children,
-  ...props
-}) => {
-  return (
-    <>
-      <Box
-        as="span"
-        aria-hidden="true"
-        sx={merge<BetterSystemStyleObject>(
-          {
-            display: 'inline-block',
-            padding: '2px 5px',
-            fontSize: 0,
-            fontWeight: 'bold',
-            lineHeight: 'condensedUltra',
-            borderRadius: '20px',
-            backgroundColor: scheme === 'primary' ? 'neutral.emphasis' : 'neutral.muted',
-            color: scheme === 'primary' ? 'fg.onEmphasis' : 'fg.default',
-            '&:empty': {
-              display: 'none',
+const CounterLabel = forwardRef<HTMLSpanElement, CounterLabelProps>(
+  ({scheme = 'secondary', sx = defaultSxProp, children, ...props}, forwardedRef) => {
+    return (
+      <>
+        <Box
+          aria-hidden="true"
+          sx={merge<BetterSystemStyleObject>(
+            {
+              display: 'inline-block',
+              padding: '2px 5px',
+              fontSize: 0,
+              fontWeight: 'bold',
+              lineHeight: 'condensedUltra',
+              borderRadius: '20px',
+              backgroundColor: scheme === 'primary' ? 'neutral.emphasis' : 'neutral.muted',
+              color: scheme === 'primary' ? 'fg.onEmphasis' : 'fg.default',
+              '&:empty': {
+                display: 'none',
+              },
             },
-          },
-          sx,
-        )}
-        {...props}
-      >
-        {children}
-      </Box>
-      <VisuallyHidden>&nbsp;({children})</VisuallyHidden>
-    </>
-  )
-}
+            sx,
+          )}
+          {...props}
+          as="span"
```

Comment:
we don't pass as through (in ts land) so avoiding it getting passed through by non-ts code